express-notes-api/index.js

```diff
@@ -0,0 +1,105 @@
+const express = require('express');
+const data = require('./data');
+const fs = require('fs');
+const app = express();
+
+const parse = express.json();
+app.use(parse);
+
+app.get('/api/notes', (req, res) => {
+  res.status(200);
+  let reply = [];
+  for (let key in data.notes) {
+    reply.push(data.notes[key]);
+  }
+  return res.send(reply);
+})
+
+app.get('/api/notes/:id/', (req, res) => {
+  let reply = [];
+  for (let key in data.notes) {reply.push(key);}
+  if (req.params.id < 0 || isNaN(req.params.id)) {
+    res.status(400)
+    return res.json({"error": "id must be a positive integer."});
+  }
```

Comment:
`400 Bad Request` should also trigger on fractional `id`s.

---

![image](https://user-images.githubusercontent.com/16143771/129801347-4c5310d9-6176-495b-89ca-39984162aae1.png)